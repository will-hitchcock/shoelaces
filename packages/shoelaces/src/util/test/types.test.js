import {
  string, bool, number, oneOf, unit,
} from '../types';

describe('string', () => {
  it('validates whether "string" is a string', () => {
    expect(string('string')).toMatchSnapshot();
  });

  it('validates whether true is a string', () => {
    expect(string(true)).toMatchSnapshot();
  });

  it('validates whether 12 is a string', () => {
    expect(string(12)).toMatchSnapshot();
  });
});

describe('bool', () => {
  it('validates whether "string" is a bool', () => {
    expect(bool('string')).toMatchSnapshot();
  });

  it('validates whether true is a bool', () => {
    expect(bool(true)).toMatchSnapshot();
  });

  it('validates whether 12 is a bool', () => {
    expect(bool(12)).toMatchSnapshot();
  });
});

describe('number', () => {
  it('validates whether "string" is a number', () => {
    expect(number('string')).toMatchSnapshot();
  });

  it('validates whether true is a number', () => {
    expect(number(true)).toMatchSnapshot();
  });

  it('validates whether 12 is a number', () => {
    expect(number(12)).toMatchSnapshot();
  });
});

describe('oneOf', () => {
  it('validates whether "ham" is either "ham" or "cheese"', () => {
    expect(oneOf(['ham', 'cheese'])('ham')).toMatchSnapshot();
  });

  it('validates whether "mayo" is either "ham" or "cheese"', () => {
    expect(oneOf(['ham', 'cheese'])('mayo')).toMatchSnapshot();
  });
});

describe('unit', () => {
  it('validates whether "100px" is a unit', () => {
    expect(unit('100px')).toMatchSnapshot();
  });

  it('validates whether "auto" is a unit', () => {
    expect(unit('auto')).toMatchSnapshot();
  });

  it('validates whether 0 is a unit', () => {
    expect(unit(0)).toMatchSnapshot();
  });

  it('validates whether "0" is a unit', () => {
    expect(unit('0')).toMatchSnapshot();
  });

  it('validates whether "90" is a unit', () => {
    expect(unit('90')).toMatchSnapshot();
  });

  it('validates whether 90 is a unit', () => {
    expect(unit(90)).toMatchSnapshot();
  });
});
