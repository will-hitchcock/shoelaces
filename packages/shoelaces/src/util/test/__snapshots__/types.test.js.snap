// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bool validates whether "string" is a bool 1`] = `false`;

exports[`bool validates whether 12 is a bool 1`] = `false`;

exports[`bool validates whether true is a bool 1`] = `true`;

exports[`number validates whether "string" is a number 1`] = `false`;

exports[`number validates whether 12 is a number 1`] = `true`;

exports[`number validates whether true is a number 1`] = `false`;

exports[`oneOf validates whether "ham" is either "ham" or "cheese" 1`] = `true`;

exports[`oneOf validates whether "mayo" is either "ham" or "cheese" 1`] = `false`;

exports[`string validates whether "string" is a string 1`] = `true`;

exports[`string validates whether 12 is a string 1`] = `false`;

exports[`string validates whether true is a string 1`] = `false`;

exports[`unit validates whether "0" is a unit 1`] = `true`;

exports[`unit validates whether "90" is a unit 1`] = `false`;

exports[`unit validates whether "100px" is a unit 1`] = `true`;

exports[`unit validates whether "auto" is a unit 1`] = `true`;

exports[`unit validates whether 0 is a unit 1`] = `true`;

exports[`unit validates whether 90 is a unit 1`] = `false`;
